src/service/video-manager-impl.js

```diff
@@ -436,6 +540,136 @@ class VideoEntry {
   }
 
   /**
+   * Called when the video's position in the viewport changed.
```

Comment:
mention that PositionObserver calls this at most once per animation frame so ppl don't freakout over the performance of this :)